src/components/Home.jsx

```diff
@@ -1,25 +1,74 @@
 import { fetchAllArticles, fetchArticlesByTopic } from "../api";
 import { useEffect, useState } from "react";
-import { useParams } from "react-router-dom";
+import { useParams, useSearchParams } from "react-router-dom";
 import { Link } from "react-router-dom";
 
 const Home = () => {
   const [allArticles, setAllArticles] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams({});
+  const [sortBy, setSortBy] = useState("created_at");
+  const [order, setOrder] = useState("DESC");
   const { topic_name } = useParams();
 
   useEffect(() => {
     if (!topic_name) {
-      fetchAllArticles().then((fetchedArticles) => {
+      fetchAllArticles(sortBy, order).then((fetchedArticles) => {
         setAllArticles(fetchedArticles.articles);
       });
     } else
       fetchArticlesByTopic(topic_name).then((topic) => {
         setAllArticles(topic.articles);
       });
-  }, [topic_name]);
+  }, [topic_name, sortBy, order]);
```

Comment:
The useEffect is using the values held in state rather than from the URL. This means that if we refresh, the URL won't match the view. You can use getters to retrieve the currents values from searchParams. Equally you can listen for a change to searchParams as a trigger for the useEffect